### PR TITLE
Update trivy

### PR DESCRIPTION
**Pull request by moukoublen**
### Summary of your changes

Update `aquasecurity/trivy` and fix API changes. 

`github.com/aquasecurity/trivy` `v0.46.1` => `v0.48.3`

Needed also major version update:
`github.com/aws/aws-sdk-go-v2/service/elasticloadbalancingv2`  `v1.19.11` => `v1.21.3`

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

#### Introducing a new rule?

- [ ] Generate rule metadata using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rules-metadata)
- [ ] Add relevant unit tests
- [ ] Generate relevant rule templates using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rule-templates), and open a PR in [elastic/packages/cloud_security_posture](https://github.com/elastic/integrations/tree/main/packages/cloud_security_posture)

### CODE REVIEW

File: vulnerability/worker_test.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ type runnerMock struct {
 }
 
 func (s runnerMock) ScanVM(_ context.Context, opts flag.Options) (trivy_types.Report, error) {
-	writer, err := os.OpenFile(opts.ReportOptions.Output, os.O_APPEND|os.O_WRONLY, 0644)
+	writer, err := os.OpenFile(opts.ReportOptions.Output, os.O_APPEND|os.O_WRONLY, 0o644)
```
Comment (amirbenun) [inline vulnerability/worker_test.go:51]: what's the difference?

**Review by amirbenun — APPROVED** [review]

**Comment by amirbenun** [comment]
Did you test it locally?

**Review by romulets — APPROVED** [review]

File: vulnerability/worker_test.go (hunk 1 above)
Comment (moukoublen) [reply vulnerability/worker_test.go:51]: No difference, just the golang language server from vscode extension updated it automatically

**Comment by orestisfl** [comment]
This is a risky update since the CNVM CI is not working, should we postpone it until we fix that?

**Comment by orouz** [comment]
> This is a risky update since the CNVM CI is not working, should we postpone it until we fix that?

i kinda hoped this would fix it, but @moukoublen just tested this and got the same error we have in the CI test:

```
VulnerabilityScanner.scan.ScanVM error: scan error: scan failed: failed analysis: EBS open error: EBS error: not found
```

**Comment by moukoublen** [comment]
@orestisfl @amirbenun 

As @orouz said, since it's not resolving the issue, I agree let's wait til we fix the cnvm ci to merge it.

**Pull request merged**